src/test/kotlin/com/charleskorn/kaml/YamlNoDefaults.kt

```diff
@@ -0,0 +1,85 @@
+/*
+
+   Copyright 2019 Jonathan Bisson.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+package com.charleskorn.kaml
+
+import ch.tutteli.atrium.api.cc.en_GB.notToBe
+import ch.tutteli.atrium.api.cc.en_GB.toBe
+import ch.tutteli.atrium.verbs.assert
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import kotlinx.serialization.Serializable
+
+object YamlNoDefaults : Spek({
+    describe("a YAML default value") {
+        describe("testing equivalence") {
+            @Serializable
+            data class LocationLayerDefault(val line: Int, val column: Int, val layer: Int? = null)
```

Comment:
Are `line` and `column` necessary here and in `LocationNoDefault`? In order to prove that the default handling behaviour is working correctly, `layer` is the only property that's necessary.